library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -227,6 +227,7 @@ private void shutDown() {
         Log.d("Shutting down service");
         getContentResolver().unregisterContentObserver(downloadManagerContentObserver);
         downloadScanner.shutdown();
+        executor.shutdownNow();
```

Comment:
yeah, actually given I've added the fix you made into my branch, I'll see if that fixes it because that could have been the issue

(but we should shutdown the executor anyway, its good practise)